src/library/scala/runtime/Statics.java

```diff
@@ -60,8 +60,10 @@ public static int doubleHash(double dv) {
   }
 
   public static int floatHash(float fv) {
+    // https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.3
+    // when fv is too large, it becomes Int.MaxValue
     int iv = (int)fv;
-    if (iv == fv)
+    if ((long)iv == (long)fv && (iv == fv))
```

Comment:
If we are always going to compute the `long` value (is that okay for Scala.js?), then we should just do it up front: `long lv = (long) fv; if (lv == fv) return longHash(lv);` or the equivalent.

On the other hand, it might just be better to pick out the particular case that is causing problems: `if (iv == fv && fv != 2.14748365E9) return iv;`.

Reply:
Don't worry about Scala.js for this specific PR. It has [its own implementation of `Statics`](https://github.com/scala-js/scala-js/blob/v1.0.0-M3/library-aux/src/main/scala/scala/runtime/Statics.scala). Actually, Scala.js is already correct: https://scalafiddle.io/sf/r7BlrZq/0 :stuck_out_tongue: 

> then we should just do it up front: `long lv = (long) fv; if (lv == fv) return longHash(lv);` or the equivalent.

Another possibility would be, now that I think of it, to use the exact same algorithm used by Scala.js. Which is that `floatHash` simply always delegates to `doubleHash`, and remove the test for `if (fv == dv)` in `doubleHash`:
```java
  public static int doubleHash(double dv) {
    int iv = (int)dv;
    if (iv == dv)
      return iv;

    long lv = (long)dv;
    if (lv == dv)
      return java.lang.Long.hashCode(lv);

    return java.lang.Double.hashCode(dv);
  }

  public static int floatHash(float fv) {
    return doubleHash((double)fv);
  }
```
That has least one advantage, which is that it reduces the worst-case length of the `doubleHash` algorithm. It does push `java.lang.Float.hashCode(fv)` and `fv.##` further away from each other, but they were already not in sync for integral values, so it's not making the situation fundamentally worse.

Reply:
Comparing with Double felt icky to I updated the commit with `(iv == fv && iv != Integer.MAX_VALUE)`.